### PR TITLE
Added latching buttons (Android)

### PR DESCRIPTION
**Pull request by ThunderousEcho**
Added latching buttons, buttons which turn on when you press them and off when you press them again.
This is necessary to play games which require you to press more than 2 buttons at once.

### CODE REVIEW

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
This is something I think will make a bunch of users happy :)

I'm planning on adding a system where you can add custom buttons to the touch overlay that you then can map to whatever you want using Dolphin's input binding system. By then using the `toggle` function, you would be able to achieve essentially the same functionality, but... Well, that way of doing things is a *lot* less discoverable for users, so I think adding a latching system like this would be nice even if it technically will be duplicated functionality eventually.

Didn't find anything in the code that needed commenting on, but let's see if @t895 would also like to review this.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.kt [hunk 1]
```diff
@@ -514,6 +515,85 @@ class EmulationActivity : AppCompatActivity(), ThemeProvider {
...
+                      R.array.wiimoteLatchableButtons, wiimoteLatchingButtons
+                ) { _: DialogInterface?, indexSelected: Int, isChecked: Boolean ->
+                    BooleanSetting.valueOf(wiimoteSettingBase + indexSelected)
+                        .setBoolean(settings, isChecked)
+                }
+            }
+        }
+
+        builder
+            .setPositiveButton(R.string.ok) { _: DialogInterface?, _: Int -> emulationFragment?.refreshInputOverlay() }
```
Comment (t895) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.kt:593]:
```suggestion
            .setPositiveButton(R.string.ok) { _: DialogInterface?, _: Int ->
                emulationFragment?.refreshInputOverlay()
            }
```

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.kt [hunk 2]
```diff
@@ -514,6 +515,85 @@ class EmulationActivity : AppCompatActivity(), ThemeProvider {
...
+            InputOverlay.OVERLAY_WIIMOTE_NUNCHUK -> {
+                val nunchukLatchingButtons = BooleanArray(9)
+                val nunchukSettingBase = "MAIN_BUTTON_LATCHING_WII_"
+
+                // For OVERLAY_WIIMOTE_NUNCHUK, settings index 7 is the D-Pad (which cannot be
+                // latching). C and Z (settings indices 8 and 9) need to map to multichoice array
+                // indices 7 and 8 to avoid a gap.
+                fun translateToSettingsIndex(idx: Int): Int {
+                    return if (idx >= 7) idx + 1 else idx
+                }
```
Comment (t895) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.kt:561]:
```suggestion
                fun translateToSettingsIndex(idx: Int): Int = if (idx >= 7) idx + 1 else idx
```

**Review by t895 — COMMENTED** [review]
Just a couple codestyle nitpicks. Otherwise, LGTM

**Comment by JosJuice** [comment]
Please squash the commits into one.

**Comment by JosJuice** [comment]
Sorry, I meant all the commits.

**Comment by t895** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**